extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/HealthBuildTimeConfig.java

```diff
@@ -0,0 +1,15 @@
+package io.quarkus.smallrye.health.runtime;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "health")
+public class HealthBuildTimeConfig {
+    /**
+     * Whether or not extensions published health check should be enabled.
+     */
+    // This config item is used by extensions to avoid to publish the health check but not directly used in the health
+    // extension. It must exists to avoid a log WARN of 'Unrecognized configuration key'.
+    @ConfigItem(name = "extensions.enabled", defaultValue = "true")
```

Comment:
How does renaming this to `enableExtensionCheck` with the config key being `enable-extension-check` sound? 

Will defaulting to hyphenated config key name  be a viable naming? e.g taking for example the present variable naming i.e `extensionsEnabled ` the configuration key will be `extensions-enabled`?

Reply:
These are the configuration names proposed in the original issue ... I'm open to suggestion but I found them easy to use and honestly I'm not a big fan of hyphenated names ...